packages/rule-tester/src/RuleTester.ts

```diff
@@ -929,130 +940,161 @@ export class RuleTester extends TestFramework {
-                      expectedSuggestion.messageId,
-                    ),
-                    `${suggestionPrefix} Test has invalid messageId '${expectedSuggestion.messageId}', the rule under test allows only one of ${friendlyIDList}.`,
-                  );
-                  assert.strictEqual(
-                    actualSuggestion.messageId,
-                    expectedSuggestion.messageId,
-                    `${suggestionPrefix} messageId should be '${expectedSuggestion.messageId}' but got '${actualSuggestion.messageId}' instead.`,
-                  );
-
-                  const unsubstitutedPlaceholders =
-                    getUnsubstitutedMessagePlaceholders(
-                      actualSuggestion.desc,
-                      rule.meta.messages[expectedSuggestion.messageId],
-                      expectedSuggestion.data,
+              } else if (Array.isArray(error.suggestions)) {
+                assert.strictEqual(
+                  messageSuggestions.length,
+                  error.suggestions.length,
+                  `Error should have ${error.suggestions.length} suggestions. Instead found ${messageSuggestions.length} suggestions`,
+                );
+
+                error.suggestions.forEach(
+                  (expectedSuggestion: SuggestionOutput<MessageIds>, index) => {
+                    assert.ok(
+                      typeof expectedSuggestion === 'object' &&
+                        expectedSuggestion != null,
+                      "Test suggestion in 'suggestions' array must be an object.",
+                    );
+                    Object.keys(expectedSuggestion).forEach(propertyName => {
+                      assert.ok(
+                        SUGGESTION_OBJECT_PARAMETERS.has(propertyName),
+                        `Invalid suggestion property name '${propertyName}'. Expected one of ${FRIENDLY_SUGGESTION_OBJECT_PARAMETER_LIST}.`,
+                      );
+                    });
+
+                    const actualSuggestion = messageSuggestions[index];
+                    const suggestionPrefix = `Error Suggestion at index ${index}:`;
+
+                    const unsubstitutedPlaceholders =
+                      getUnsubstitutedMessagePlaceholders(
+                        actualSuggestion.desc,
+                        rule.meta.messages[expectedSuggestion.messageId],
+                        expectedSuggestion.data,
+                      );
+
+                    assert.ok(
+                      unsubstitutedPlaceholders.length === 0,
+                      `The message of the suggestion has ${unsubstitutedPlaceholders.length > 1 ? `unsubstituted placeholders: ${unsubstitutedPlaceholders.map(name => `'${name}'`).join(', ')}` : `an unsubstituted placeholder '${unsubstitutedPlaceholders[0]}'`}. Please provide the missing ${unsubstitutedPlaceholders.length > 1 ? 'values' : 'value'} via the 'data' property for the suggestion in the context.report() call.`,
```

Comment:
[Refactor] This change doesn't seem to be related to the upstream PR. Is there any reason to move this peace of code (`unsubstitutedPlaceholders` + `assert.ok(unsubstitutedPlaceholders.length === 0, ...`) before the ` if (hasOwnProperty(expectedSuggestion, 'desc')) {` line?